### ISSUE TITLE
The already-solved "check mark" should be displayed everywhere a Kata name is shown

### ISSUE THREAD

**Issue by dinglemouse2250** [Codewars/codewars.com#752]
The "check mark" indicating if the Kata has already been solved is really useful, but it is not ubiquitous.

There are many cases where Kata are listed but the "check mark" is not there. So often I am pressing the "Train Again" button just to figure out - *did I complete this one yet or not?*

**Comment by jhoffner** [comment]
Yeah this has been mentioned before, this might be a duplicate. I definitely see the usefulness, though I must admit it worries me when I see people get upset about doing a kata more than once. One of the major points of kata is to do them multiple times in order to get better/faster at the types of coding patterns that are used to solve them. With that said - not enough of an effort was made from a design/gamification stand-point to encourage this type of behavior, so I guess I shouldn't be too surprised. 

For what its worth though. From the info/solutions/discourse pages, the button will say "Train" instead of "Train Again" if you haven't attempted it yet. Not always useful though since you may have skipped over it previously.

**Comment by suic86** [comment]
@jhoffner:

> One of the major points of kata is to do them multiple times in order to get better/faster at the types of coding patterns that are used to solve them.

Let me have a heretic comment:

If this is one of the goals of CW, why are duplicate katas considered to be (so) wrong (evil, harmful, degrading, ...). In principle solving 3 duplicate katas in three different ways is kind of similar to the quoted idea.
Creating a simple duplicate kata (or its translation) _having proper quality_ is a good exercise for the author or translator (e. g. it's a good way to familiarize with random input generation in different language or in case of Haskell even with functors and _monads_(!)). I'm far from being egalitarian but I don't see to point of "punishing" people who create their own simple katas just because these are duplicates as someone else joined CW earlier them and created a kata.\*
On the other hand, I understand that kata duplication on CW isn't a black and white question and see the possible drawbacks of being to benevolent. However I think that here we need more common sense than strictness and also some "definition" of what is/n't considered to be a duplicate.

Regards,

suic

\* Here comes the often used "create rather a translation" argument. But e. g. What about people who know only one programming language? What if the kata is translated only to language which one doesn't now? Why would one create a translation which will never get approved? Last but not least: Coders in general are lazy and don't like admin stuff. From this POV is understandable that one rather creates u duplicate than spend time with "duplication hunting" or finding an existing kata with a similar description.

**Comment by dinglemouse2250** [comment]
> For what its worth though. From the info/solutions/discourse pages, the button will say "Train" instead of "Train Again" if you haven't attempted it yet. 

Yes. I think I was a bit vague in my original comment, In the many lists like ("Similar Kata:") and ("Other Kata by XXX") and a users ("Kata>Completed") and ("Kata>Authored") all you have is the *title*. So cannot even see Train/Train-Again unless click thru the title.

**Comment by anter69** [comment]
Related: #727, #1162, #1272

**Comment by ivanji** [comment]
For UX reasons, there should be better ways to identify when a Kata has been completed, so far I can only realise it when the 'Train again' button appears. I don't even see the check mark the OP mentioned.

Please update the styles or include more cues so we could easily identify them. Thanks.
